docs/profiler.md

```diff
@@ -0,0 +1,122 @@
+`--nonces` The more nonces one does in ONE data pass, the bigger chance to find a proof. The downside is the more `--nonces` the more work CPU will need to do.
+
+To simplify the estimation of nonces please use the following formula:
+in google sheets format
+```
+1-(1-(1-BINOM.DIST(36;10^9;26/10^9;TRUE)))^{put nonces here}
+```
+in wolfram alpha format
+```
+1-(1-(1-CDF[BinomialDistribution[10^9, 26/10^9],36]))^{put nonces here}
+```
+
+That formula will output a probability of finding proof in one data pass. Proving part is written in a way that it will do multiple data passes IF needed.
+
+Please note that in genreal proof speed should almost linearly scale up with the amount of cores (assuming that they're equally fast), but in the same time it will linearlly scale down with the amount of nonces in groups of 16 (32 nonces should be twice as slow as 16)
+
+## How to interpret the results
+
+```
+./profiler --data-size 1 --threads=1 --data-file data.bin --nonces=64
+{
+  "time_s": 12.09140029,
+  "speed_gib_s": 0.41351703525481415
+}
+```
+
+Here we see the following
+* 1GiB file `data.bin` was used for the benchmark
+* 1 thread was used
+* 64 nonces were used
+
+The benchmark took 12.09 seconds to complete and the speed was 0.41GiB/s.
+But from previous formula we know that the probability of finding a proof with 64 nonces is 79.39%. Therefore there is 20% chance that we will need to read the data again. Therefore the actual time can be significanly longer.
+```
+./profiler --data-size 1 --threads=1 --data-file data.bin --nonces=128
+{
+  "time_s": 13.152850458,
+  "speed_gib_s": 0.22808744078552953
+}
+```
+
+We see here that the speed dropped by half. That clearly indicates that CPU is our limiting factor.
+
+With 128 nonces there is 95.75% chance to find a proof in one data pass.
+
+```
+./profiler --data-size 1 --threads=10 --data-file data.bin --nonces=128
+{
+  "time_s": 10.331206291,
+  "speed_gib_s": 1.8390882405040923
+}
+```
+
+With 10 threads we can see that speed increased by 8 times. But again it's important to remember that we don't necessarily need to optimize for the speed, we need to optimize for the probability of finding a proof in time. There are 12 hours to find and submit proof.
+
+Based on these outputs you need to decide what is the best configuration for your hardware. Please note that the speed of the proof generation is not the only factor.
+
+## Is that all that is happening during the proof generation?
+Actually no, there is one more item which is that for each group of 16 nonces there is computation required. That exists to prevent a certain kind of exploit where an adversary with cheap processing power can replace some storage with much more computation, we’ve introduced a small amount of additional computation to the PoST proving process making this attack too expensive to carry out.
```

Comment:
I would shorten the whole sentence to something like:

```
Additionally for every group of 16 nonces there is a proof of work - often referred to as `k2pow` - required. It serves as mitigation against some possible attacks by dishonest smeshers.
```

Reply:
I took it from the annoucement that was given tot the community, wanted to make it clear that it's not a PoW per see but kinda protection against attack.

Reply:
Then instead of "there is a proof of work" make it "there is an additional computation" 🙂

Reply:
Changed.